tests/Resources/Page/ODH/ODHDashboard/ODHDashboardAPI.resource

```diff
@@ -0,0 +1,65 @@
+    ...                 on the given token (i.e., user)
+    [Arguments]     ${endpoint}     ${token}    ${json_body}
+    ${headers}=    Create Dictionary     Cookie=_oauth_proxy=${token}   Content-type=application/json
+    ${payload}=      Load Json String    json_string=${json_body}
+    ${response}=    RequestsLibrary.Put  ${ODH_DASHBOARD_URL}/${endpoint}    expected_status=any
+    ...             headers=${headers}    data=${json_body}   timeout=5  verify=${False}
+
+Perform Dashboard API Endpoint POST Call
+    [Documentation]     Runs a PUT call to the given API endpoint. Result may change based
+    ...                 on the given token (i.e., user)
+    [Arguments]     ${endpoint}     ${token}    ${json_body}
+    ${headers}=    Create Dictionary     Cookie=_oauth_proxy=${token}   Content-type=application/json
+    ${payload}=      Load Json String    json_string=${json_body}
+    ${response}=    RequestsLibrary.Post  ${ODH_DASHBOARD_URL}/${endpoint}    expected_status=any
+    ...             headers=${headers}    data=${json_body}   timeout=5  verify=${False}
+
+Perform Dashboard API Endpoint PATCH Call
+    [Documentation]     Runs a PATCH call to the given API endpoint. Result may change based
+    ...                 on the given token (i.e., user)
+    [Arguments]     ${endpoint}     ${token}    ${json_body}
+    ${headers}=    Create Dictionary     Cookie=_oauth_proxy=${token}   Content-type=application/json
+    ${payload}=     Load Json String    json_string=${json_body}
+    ${response}=    RequestsLibrary.Patch  ${ODH_DASHBOARD_URL}/${endpoint}    expected_status=any
+    ...             headers=${headers}    data=${json_body}   timeout=5  verify=${False}
+
+Perform Dashboard API Endpoint DELETE Call
+    [Documentation]     Runs a GET call to the given API endpoint. Result may change based
+    ...                 on the given token (i.e., user)
+    [Arguments]     ${endpoint}     ${token}
+    ${headers}=    Create Dictionary     Cookie=_oauth_proxy=${token}
+    ${response}=    RequestsLibrary.DELETE  ${ODH_DASHBOARD_URL}/${endpoint}   expected_status=any
+    ...             headers=${headers}   timeout=5  verify=${False}
+
+Operation Should Be Allowed
+    [Documentation]     Checks if the API call returns an HTTP code 200 (SUCCESS)
+    Run Keyword And Continue On Failure  Status Should Be  200
+
+Operation Should Be Unauthorized
+    [Documentation]     Checks if the API call returns an HTTP code 401 (Unauthorized)
+    Run Keyword And Continue On Failure  Status Should Be  401
+
+Operation Should Be Forbidden
+    [Documentation]     Checks if the API call returns an HTTP code 403 (FORBIDDEN)
+    Run Keyword And Continue On Failure  Status Should Be  403
+
+Operation Should Be Unavailable
+    [Documentation]     Checks if the API call returns an HTTP code 404 (NOT FOUND)
+    Run Keyword And Continue On Failure  Status Should Be  404
```

Comment:
I thought it was clear for understand the TC to have a kw saying explicitly "the operation should be allowed/forbidden/etc". I could change of course

Reply:
I don't have a strong preference either way to be honest, both options work fine and you've already written all TCs to use these keywords. To me it would've been a bit cleaner to have one keyword call for each request instead of two, that way also when failing you'd have all info in one place instead of split between the two, but it's not a big deal.

Reply:
okok. Let's see if there are other opinions from others. In case I can change, np

Reply:
I think both approaches are good. 

Maybe the current keyword names could be more specific (with optional parameter continue_on_failure), like:
`Verify Dashboard API Call Was Allowed`    continue_on_failure=True
`Verify Dashboard API Call Was Unauthorized`     continue_on_failure=True
`Verify Dashboard API Call Was Unavailable`     continue_on_failure=True

But having it in just one call looks god to me too:
```
Perform Dashboard API Endpoint GET Call   endpoint=${CONFIG_ENDPOINT}
...    token=${BASIC_USER_TOKEN}    expected_status=200    continue_on_failure=True
```


Another consideration is that the current keywords at ODHDashboardAPI.resource seems to be designed to test the access to the endpoints, but are not really designed to interact with the API (e.g. the GET calls don't return the results).  If it was useful to interact with the API, maybe it would make sense to refactor the code (in a future PR) to add more knowledge of the API in ODHDashboardAPI (like having there the endpoint subpaths, or deciding if a PUT or PATCH call is needed) and simplify the tests using the api like this:

```
${cluster_settings}=    ODHDashboard.Get Cluster Settings  dashboard_url=${ODH_DASHBOARD_URL}    token=${BASIC_USER_TOKEN}    expected_status=200
ODHDashboard.Set Cluster Settings  dashboard_url=${ODH_DASHBOARD_URL}    token=${BASIC_USER_TOKEN}    expected_status=200   settings_json=${CLUSTER_SETTINGS_ENDPOINT_BODY}    

${config}=    ODHDashboard.Get Config   dashboard_url=${ODH_DASHBOARD_URL}    token=${BASIC_USER_TOKEN}    expected_status=200
ODHDashboard.Set Config   dashboard_url=${ODH_DASHBOARD_URL}    token=${BASIC_USER_TOKEN}    expected_status=200    config_json=${CONFIG_ENDPOINT_BODY}        

${groups}=    ODHDashboard.Get Groups   dashboard_url=${ODH_DASHBOARD_URL}    token=${BASIC_USER_TOKEN}    expected_status=200
```

Reply:
I will change the kw names and add the suggested parameter. 
Regarding the having all in one keyword, I'll see when I finish the PR, if there's time and the effort is not big, I'll do. However, for sake of reading, I would prefer using separated kw. Let's see.

> Another consideration is that the current keywords at ODHDashboardAPI.resource seems to be designed to test the access to the endpoints, but are not really designed to interact with the API


since we have never discussed to do API testing but the preferred path was to use UI (like a user), then i didn't thought to hanlde responses. We can do in future PR if we need

Reply:
re-opening the conversation to not forget the changes here. I will do tomorrow, I was focused on finish covering all the endpoints